AirCasting/Utils/AveragingService.swift

```diff
@@ -81,39 +78,45 @@ final class AveragingService: NSObject, ObservableObject {
         }
     }
     
-    private func perform(sessionEntity: SessionEntity, averagingWindow: AveragingWindow) {
-        if let uuid = sessionEntity.uuid {
-            measurementStreamStorage.accessStorage { storage in
-                
-                let session = try? storage.getExistingSession(with: uuid)
+    private func perform(uuid: SessionUUID, averagingWindow: AveragingWindow, windowDidChange: Bool) {
+        measurementStreamStorage.accessStorage { storage in
```

Comment:
I don't know... The `accessStorage` block here is doing some heavy stuff and since we moved to a single-context (`editContext` being `lazy var`) it will block all other editing while being performed. I'd recommend avoiding this

Reply:
Could you be more specific? Like giving tips on what should I do?